src/Routing/Getter.purs

```diff
@@ -0,0 +1,123 @@
+module Routing.Getter (
+  route,
+  or,
+  contains,
+  runRoute,
+  hashes,
+  checks,
+  routes,
+  PErr(),
+  Route(),
+  RouteMsg,
+  toMsg,
+  Check(),
+  Checks()
+  ) where
+
+import Data.Either
+import Data.Maybe
+import Data.Tuple
+import Control.Monad.Eff
+import Control.Monad.State
+import Control.Monad.Trans
+import Control.Monad.State.Class
+import Control.Monad.Error
+import Control.Apply ((*>))
+import Control.Alt ((<|>))
+import Data.String.Regex (replace, noFlags, Regex(), regex)
+import Data.Array (findIndex, drop, take)
+import Data.Traversable (traverse)
+
+import qualified Data.StrMap as M
+import qualified Text.Parsing.Parser as P
+import qualified Text.Parsing.Parser.Combinators as P
+
+import Routing.Parser (parse, template, StateObj())
+
+type RouteParser = P.ParserT String StateObj Checks
+type Check = Tuple String (M.StrMap String)
+type Checks = Tuple Check [Check]
+type PErr a = Either P.ParseError a 
+newtype Route = Route RouteParser
+
+runRoute :: String -> Route -> PErr Checks
+runRoute s (Route p) =
+  case runState (P.runParserT s p) M.empty of
+    Tuple lr state -> lr
+
+             
+route :: String -> String -> PErr Route
+route name templateStr = do
+  parser <- parse <$> P.runParser templateStr template
+  let p = parser *> lift get >>= \res ->
+        pure $ Tuple (Tuple name res) []
+  pure $ Route p
+
+or :: Route -> Route -> Route
```

Comment:
ok, I'm going to add `Semiring` instance. Just need to figure out what is `zero`